### ISSUE TITLE
process_start_time_seconds not working as expected for VM servers

### ISSUE THREAD

**Issue by nilendudas1422** [VictoriaMetrics/VictoriaMetrics#767]
**Describe the bug**
`changes(process_start_time_seconds{job=~"victoriametrics", service=~"victoriametrics"}[15m])` always shows a constant value of 30 when plotted via Grafana

**To Reproduce**
Start VM Server
Scrape metrics from Vmanget and Vm server
and plot them using Grafana

**Expected behavior**
The changes function will not work as expected for the `service=~"victoriametrics"` but will work for `service=~"vmagent"`

**Screenshots**
<img width="1675" alt="Screenshot 2020-09-14 at 2 08 35 PM" src="https://user-images.githubusercontent.com/10692576/93066439-a2337f80-f697-11ea-873b-bc12e72a647f.png">

**Version**
v1.40.1

**Used command-line flags**
 --storageDataPath: "/var/lib/victoriametris"
 --selfScrapeInterval: 0
 --retentionPeriod: 6
 --memory.allowedPercent: 60

**Comment by f41gh7** [comment]
`--selfScrapeInterval: 0` - flag disables self scraping, provide positive value to it and you should be able to see `process_start_time_seconds` at VictoriaMetrics.

**Comment by valyala** [comment]
@nilendudas1422 , could you show a graph on the same time range for the following query:

```promql
process_start_time_seconds{job="victoriametrics", service="victoriametrics"}
```

**Comment by nilendudas1422** [comment]
@f41gh7 The selfScrapeInterval is 0 because the metrics for vm are being scraped via vmagent.

@valyala Upload. the screenshot as you requested
<img width="1676" alt="Screenshot 2020-09-22 at 6 03 56 PM" src="https://user-images.githubusercontent.com/10692576/93882541-ff65ab80-fcfd-11ea-99c3-67b389c92818.png">

**Comment by denisgolius** [comment]
Hi @valyala !
Could you take a look?

**Comment by just1900** [comment]
@valyala Encountering the same issue, here are some timestamp-value pairs retrieved from vmselect.  
the timestamp is changed frequently between "1690290242.5500002" and "1690290242.55"

```
                    [
                        1690293366.486,
                        "1690290242.5500002"
                    ],
                    [
                        1690293395.675,
                        "1690290242.5500002"
                    ],
                    [
                        1690293424.864,
                        "1690290242.55"
                    ],
                    [
                        1690293454.053,
                        "1690290242.55"
                    ],
                    [
                        1690293483.242,
                        "1690290242.55"
                    ],
                    [
                        1690293512.431,
                        "1690290242.55"
                    ],
                    [
                        1690293541.62,
                        "1690290242.55"
                    ],
                    [
                        1690293570.809,
                        "1690290242.55"
                    ],
                    [
                        1690293599.998,
                        "1690290242.5500002"
                    ],
                    [
                        1690293629.187,
                        "1690290242.5500002"
                    ],
                    [
                        1690293658.376,
                        "1690290242.55"
                    ],
                    [
                        1690293687.565,
                        "1690290242.55"
                    ],
                    [
                        1690293716.754,
                        "1690290242.55"
                    ],
                    [
                        1690293745.943,
                        "1690290242.55"
                    ],
                    [
                        1690293775.132,
                        "1690290242.55"
                    ],
                    [
                        1690293804.321,
                        "1690290242.55"
                    ],
                    [
                        1690293833.51,
                        "1690290242.5500002"
                    ],
                    [
                        1690293862.699,
                        "1690290242.55"
                    ],
                    [
                        1690293891.888,
                        "1690290242.55"
                    ],
                    [
                        1690293921.077,
                        "1690290242.55"
                    ],
                    [
                        1690293950.266,
                        "1690290242.55"
                    ],
                    [
                        1690293979.455,
                        "1690290242.55"
                    ],
                    [
                        1690294008.644,
                        "1690290242.55"
                    ],
                    [
                        1690294037.833,
                        "1690290242.55"
                    ],
                    [
                        1690294067.022,
                        "1690290242.55"
                    ],
                    [
                        1690294096.211,
                        "1690290242.55"
                    ],
                    [
                        1690294125.4,
                        "1690290242.55"
                    ],
                    [
                        1690294154.589,
                        "1690290242.55"
                    ],
                    [
                        1690294183.778,
                        "1690290242.55"
                    ],
                    [
                        1690294212.967,
                        "1690290242.55"
                    ],
                    [
                        1690294242.156,
                        "1690290242.5500002"
                    ],
                        
```

**Comment by pgassmann** [comment]
I also experience fluctuating start_time_seconds on 2 vmagent instances running in docker on the same server. 
latest vmagent version 1.94.0
This causes firing vmagent restart alerts with the query  `changes(process_start_time_seconds{job=~"victoriametrics|vmagent|vmalert"}[15m]) > 2`
![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/460192/3aa660f5-98df-4f81-98a8-fd1eeab3aac4)

the instances are running for 38h straight now, there are no error logs or anything else that looks suscpicious. It happens on two newly created instances. all other existing 34 vmagent instances look fine.

@valyala @f41gh7 what could cause those fluctuating values?

**Comment by pgassmann** [comment]
This issue is still present with version 1.96.0. Again it happens only on two instances on one server, all other seem ok.

I have no idea how a recorded timestamp should fluctuate? Even if it was a rounding error or precision issue, it should not change. there should be no calculation, it is not uptime, it's the tiimestamp of the start time.

![grafik](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/460192/0ecc3721-c626-4924-a42f-3e4a5719b2ae)

**Comment by pgassmann** [comment]
here's the history over 4 days.
the two fluctuating vmagents are in docker, the one on the host has straight lines between restarts.
![grafik](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/460192/34d25d48-c70d-420e-9451-be21d69f5b72)

**Comment by pgassmann** [comment]
haha. found my issue. The two vmagent instances in docker were in two separate compose projects, but connected to the same webproxy network. they were configured identically, so they both were known with the same name in the docker network. so they randomly scraped themself or the instance from the other compose project.

Fix: put the vmagent (sidecar) in the default network and only expose the actual webservice to the webproxy network.

**Comment by valyala** [comment]
The issue with incorrect calculations for [changes](https://docs.victoriametrics.com/metricsql/#changes) function should be fixed in the commit 0cf56c1ba5b3efb76e684f476cc1f4e7ac3483e1 . This commit will be included in the next release.

In the mean time it is possible to verify the bugfix by building VictoriaMetrics from source code:

- Build single-node VictoriaMetrics from the commit 0cf56c1ba5b3efb76e684f476cc1f4e7ac3483e1 according to [these docs](https://docs.victoriametrics.com/#how-to-build-from-sources).
- Build cluster components of VictoriaMetrics from the commit def5573f9204508fba028ee0df200eef806e0bfc according to [these docs](https://docs.victoriametrics.com/cluster-victoriametrics/#building-from-sources).

**Comment by valyala** [comment]
VictoriaMetrics should properly handle small fluctuations in [raw samples](https://docs.victoriametrics.com/keyconcepts/#raw-samples), which could lead to incorrect `changes()` results, starting from [v1.98.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue as fixed then.

**Comment by valyala** [comment]
FYI, the bugfix has been also included in [v1.97.2 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.97.2).

**Comment by valyala** [comment]
The bugfix has been also included in [v1.93.12 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.93.12).
